test_general/api/test_app.py

```diff
@@ -144,3 +146,82 @@ def test_general_api_returns_500_bad_pdf():
     assert response.json() == {"detail": f"{tmp.name} does not appear to be a valid PDF"}
     assert response.status_code == 400
     tmp.close()
+
+
+def test_parallel_mode_correct_result():
+    """
+    Validate that splitting up the pages and sending to the api
+    returns the same result as usual
+    Note that this test makes calls to prod for now
```

Comment:
Not a huge fan of tests that hit prod. Are we doing this elsewhere? Any alternatives?

Reply:
Hmm, to your other point, I guess we can override the url in this case to point it back to itself. I considered mocking it at first, but we want to validate the results, so it feels artificial to be testing against whatever value we put in there.

Reply:
Okay, I've mocked it to switch out the api call for `partition()` which allows us to test merging the results without dependencies.

Reply:
Awesome, thanks. Yea, this makes sense to me!

Reply:
Awesome, thanks. Yea, this makes sense to me!